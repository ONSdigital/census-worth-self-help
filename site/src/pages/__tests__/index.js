import React from "react"
import renderer from "react-test-renderer"
import Index from "../index"
import { render } from "react-testing-library"

describe("Index", () => {
  it("renders correctly without data and has no alert", () => {
    // check snapshot
    const empty_data = {}
    const tree = renderer.create(<Index data={empty_data}/>).toJSON()
    expect(tree).toMatchSnapshot()

    // check no alert
    const { queryByTestId } = render(<Index data={empty_data}/>)
    expect(queryByTestId('alert-message')).toBeNull()
  })

  it("renders correctly with alert data and has alert", () => {
    // check snapshot
    const alert_data = { markdownRemark : { frontmatter: {alert_content: "this is test content"}}}
    const tree = renderer.create(<Index data={alert_data}/>).toJSON()
    expect(tree).toMatchSnapshot()
    
    // check alert content
    const { getByTestId } = render(<Index data={alert_data}/>)
    expect(getByTestId("alert-message")).toHaveTextContent("Alert: this is test content")
  })
})