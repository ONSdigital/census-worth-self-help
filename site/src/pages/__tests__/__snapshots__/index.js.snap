// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly with alert data and has alert 1`] = `
<div>
  <div
    data-testid="alert-message"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  >
    <b>
      Alert
    </b>
    :Â 
    <span>
      this is test content
    </span>
  </div>
  <p>
    hello world
  </p>
</div>
`;

exports[`Index renders correctly without data and has no alert 1`] = `
<div>
  <p>
    hello world
  </p>
</div>
`;
